From e067643b4e321db4de3274198b13732cce37f5c1 Mon Sep 17 00:00:00 2001
From: Sebastian Kropatsch <seb-dev@mail.de>
Date: Wed, 12 Jun 2024 22:48:13 +0200
Subject: [PATCH 4/5] arm64: dts: rockchip: Enable lower USB3 port on NanoPi
 R6C/R6S

Enable support for the lower USB 3.0 Type-A port on the NanoPi R6C and
NanoPi R6S. The upper USB 2.0 Type-A port is already supported.

Signed-off-by: Sebastian Kropatsch <seb-dev@mail.de>
---
 .../boot/dts/rockchip/rk3588s-nanopi-r6.dtsi  | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

--- a/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-nanopi-r6.dtsi
@@ -740,6 +740,17 @@
 	status = "okay";
 };
 
+/* USB2 PHY for USB 3.0 Type-A (lower port)*/
+&u2phy0 {
+	status = "okay";
+};
+
+&u2phy0_otg {
+	phy-supply = <&vcc5v0_usb_otg0>;
+	status = "okay";
+};
+
+/* USB2 PHY for USB 2.0 Type-A (upper port)*/
 &u2phy2 {
 	status = "okay";
 };
@@ -755,10 +766,27 @@
 	status = "okay";
 };
 
+/* USB 2.0 Type-A (upper port) */
+/* PHY: <&u2phy2_host> */
 &usb_host0_ehci {
 	status = "okay";
 };
 
+/* USB 2.0 Type-A (upper port) */
+/* PHY: <&u2phy2_host> */
 &usb_host0_ohci {
 	status = "okay";
 };
+
+/* USB 3.0 Type-A (lower port) */
+/* PHYs: <&u2phy0_otg>, <&usbdp_phy0> */
+&usb_host0_xhci {
+	dr_mode = "host";
+	extcon = <&u2phy0>;
+	status = "okay";
+};
+
+/* USB3 PHY for USB 3.0 Type-A (lower port)*/
+&usbdp_phy0 {
+	status = "okay";
+};
